pkg/apis/options/login_url_parameters.go

```diff
@@ -0,0 +1,59 @@
+package options
+
+import (
+	"regexp"
+
+	"github.com/oauth2-proxy/oauth2-proxy/v7/providers"
+)
+
+// LoginURLParameter is the configuration for a single query parameter that
+// can be passed through from the `/oauth2/start` endpoint to the IdP login
+// URL.  The "default" option specifies the default value or values (if any)
+// that will be passed to the IdP for this parameter, and "allowed" is a
+// regular expression specifying which values (if any) that a caller passes
+// in the query string to `/oauth2/start` are permitted to override the defaults.
+// If _only_ a default is specified and no "allowed" then the parameter is
+// effectively fixed - the default value will always be used and anything
+// passed to the start URL will be ignored.  If _only_ "allowed" is specified
+// but no default then the parameter will only be passed on to the IdP if
+// the caller provides it, and no value will be sent otherwise.  Note that
+// the "allowed" regex is implicitly anchored with a `^` at the start and `$`
+// at the end, so it must match the _whole_ of the value to be considered
+// valid.
+type LoginURLParameter struct {
+	// Default specifies a default value or values that will be
+	// passed to the IdP if not overridden.
+	Default []string `json:"default,omitempty"`
+
+	// Allowed specifies a regular expression to validate any
+	// override values passed to the start endpoint.
+	Allowed *string `json:"allowed,omitempty"`
+}
+
+// ConvertLoginURLParameters configures a provider's LoginURLParameters from
+// the specified values taken from the options.  Any errors that occur while
+// parsing the regular expressions in the configuration will be appended to
+// the msgs slice.  Ideally this function would be a method of ProviderData
+// but that creates a circular import dependency.
+func ConvertLoginURLParameters(p *providers.ProviderData, paramsFromOptions map[string]LoginURLParameter, msgs []string) []string {
```

Comment:
Can this be moved out of here to avoid having to import the providers package? Probably best to put it within the providers package or providers/util maybe